py-simpy: New package

alex391 opened the pull request:
<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->

sidpbury committed:
pypi build of py-simpy

qwertos committed:
[py-simpy] toml explicitly called out

alex391 committed:
py-simpy: New version

alex391 commented:
How should I fix this gitlab-ci error?

alex391 commented:
@spackbot re-run pipeline

alex391 committed:
py-setuptools-scm: Added new version

wdconinc commented on var/spack/repos/builtin/packages/py-simpy/package.py:
```diff
@@ -0,0 +1,21 @@
+    homepage = "https://simpy.readthedocs.io/"
+    pypi = "simpy/simpy-4.0.2.tar.gz"
+
+    version("4.1.1", sha256="6d8adc0229df6b02fb7e26dcd1338703b4f4f63f167a5ac2a7213cb80aba4484")
```
```suggestion
    version("4.1.1", sha256="06d0750a7884b11e0e8e20ce0bc7c6d4ed5f1743d456695340d13fdff95001a6")
```

alex391 replied:
Where did you get that hash? I found `6d8adc0229...` from https://pypi.org/project/simpy/#copy-hash-modal-bf5289a4-c487-45af-a0c3-fb11fa764679

wdconinc replied:
![image](https://github.com/spack/spack/assets/4656391/4271d360-68a8-4536-928d-d2c22b675648)

wdconinc replied:
But also `spack checksum py-simpy 4.1.1` should return the version line to use. You had the exact same hash for 4.1.1 and 4.0.2, so that was suspicious.

alex391 replied:
Oh! I can't read! Yeah, you are correct, the hash I have is wrong. Will fix.

wdconinc commented on var/spack/repos/builtin/packages/py-setuptools-scm/package.py:
```diff
@@ -10,11 +10,12 @@ class PySetuptoolsScm(PythonPackage):
 
     homepage = "https://github.com/pypa/setuptools_scm"
-    pypi = "setuptools_scm/setuptools_scm-4.1.2.tar.gz"
+    pypi = "setuptools-scm/setuptools-scm-8.0.4.tar.gz"
```
This breaks older version installs since the URL is not derived correctly.

wdconinc replied:
Not canonical spack, but this does the job:
```py
    def url_for_version(self, version):
        if version >= Version(8):
            return super().url_for_version(version).replace("_", "-")
        else:
            return super().url_for_version(version)
```

wdconinc replied:
"You could call super once and store it in a variable." https://spackpm.slack.com/archives/C5VL7V81G/p1714549700619169?thread_ts=1714517377.256679&cid=C5VL7V81G

alex391 committed:
py-setuptools-scm: add url_for_version

Because versions @:7 have an underscore (setuptools_scm) in the URL, and
newer versions have a dash (setuptools-scm).

alex391 committed:
py-setuptools-scm: fix flake8 line too long

alex391 committed:
py-simpy: Fix hash

wdconinc reviewed (approved):
Confirmed install on two new py-simpy versions, and with setuptools 7.1.0 as well for the older version:
```
# spack find -lv py-simpy
-- linux-ubuntu20.04-x86_64_v3 / gcc@11.4.0 ---------------------
ntclwdq py-simpy@4.0.2 build_system=python_pip  xoniath py-simpy@4.0.2 build_system=python_pip  pcqqlwl py-simpy@4.1.1 build_system=python_pip
==> 3 installed packages
```

wdconinc commented:
Thanks for the contribution. Enabled auto-merge, but pipelines are running a bit behind from what I see on slack due to some changes in core packages.

alex391 commented:
Thank you for the review @wdconinc!
